### PR TITLE
agent: fix hook execution issues

### PR DESCRIPTION
**Pull request by liubin**
Envs contain null-byte will cause running hooks to panic,
this commit will filter envs and only pass valid envs to hooks.

Fixes: #3667

agent: handle hook process result
The current hook process is handled by just calling
unwrap() on it, sometimes it will cause panic.

Handling all Result types and checking the error can
avoid panic.

Fixes: #3649

Signed-off-by: bin <bin@hyper.sh>

### CODE REVIEW

File: src/agent/rustjail/src/container.rs [hunk 1]
```diff
@@ -1488,14 +1488,9 @@ async fn execute_hook(logger: &Logger, h: &Hook, st: &OCIState) -> Result<()> {
...
-    let env: HashMap<String, String> = h
-        .env
-        .iter()
-        .map(|e| {
-            let v: Vec<&str> = e.split('=').collect();
-            (v[0].to_string(), v[1].to_string())
-        })
-        .collect();
+
+    // all invalid envs will be ommit, only valid envs will be passed to hook.
```
Comment (jodh-intel) [inline src/agent/rustjail/src/container.rs:1492]: Typo: "omitted".

**Review by jodh-intel — APPROVED** [review]
Thanks @liubin.

lgtm

File: src/agent/rustjail/src/container.rs (hunk 1 above)
Comment (liubin) [reply src/agent/rustjail/src/container.rs]: Fixed, please re-approve, thanks.

**Review by jodh-intel — APPROVED** [review]
Thanks @liubin.

lgtm

**Review by devimc — APPROVED** [review]

**Review by sameo — APPROVED** [review]

**Pull request merged**